pip/index.py

```diff
@@ -1019,7 +1019,7 @@ def fmt_ctl_no_use_wheel(fmt_ctl):
     fmt_ctl_no_binary(fmt_ctl)
     warnings.warn(
         '--no-use-wheel is deprecated and will be removed in the future. '
-        ' Please use --no-binary :all: instead.', DeprecationWarning,
+        ' Please use --no-binary :all: instead.', RemovedInPip9Warning,
```

Comment:
It has been deprecated for several major pip versions if I recall, but I can easily change it to `RemovedInPip10Warning`.  Certainly using python's builtin DeprecationWarning means that nearly nobody will see it, which is wrong.